pkg/api/validation/cluster_validation_test.go

```diff
@@ -40,129 +43,164 @@ func TestValidateScyllaCluster(t *testing.T) {
+			},
+			expectedErrorString: `spec.datacenter.racks[1].name: Duplicate value: "test-rack"`,
 		},
 		{
-			name:    "invalid intensity in repair task spec",
-			obj:     invalidIntensity,
-			allowed: false,
+			name: "invalid intensity in repair task spec",
+			obj:  invalidIntensity,
+			expectedErrorList: field.ErrorList{
+				&field.Error{Type: field.ErrorTypeInvalid, Field: "spec.repairs[0].intensity", BadValue: "100Mib", Detail: "invalid intensity, it must be a float value"},
+			},
+			expectedErrorString: `spec.repairs[0].intensity: Invalid value: "100Mib": invalid intensity, it must be a float value`,
 		},
 		{
-			name:    "non-unique names in manager tasks spec",
-			obj:     nonUniqueManagerTaskNames,
-			allowed: false,
+			name: "invalid intensity in repair task spec && non-unique names in manager tasks spec",
+			obj:  nonUniqueManagerTaskNames,
+			expectedErrorList: field.ErrorList{
+				&field.Error{Type: field.ErrorTypeInvalid, Field: "spec.repairs[0].intensity", BadValue: "", Detail: "invalid intensity, it must be a float value"},
+				&field.Error{Type: field.ErrorTypeDuplicate, Field: "spec.backups[0].name", BadValue: "task-name"},
+			},
+			expectedErrorString: `[spec.repairs[0].intensity: Invalid value: "": invalid intensity, it must be a float value, spec.backups[0].name: Duplicate value: "task-name"]`,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			err := validation.ValidateScyllaCluster(test.obj)
-			if test.allowed {
-				if err != nil {
-					t.Errorf("Wrong value returned from checkValues function. Message: '%s'", err)
-				}
-			} else {
-				if err == nil {
-					t.Errorf("Wrong value returned from checkValues function. Message: '%s'", err)
-				}
+			errList := validation.ValidateScyllaCluster(test.obj)
+			if !reflect.DeepEqual(errList, test.expectedErrorList) {
+				t.Errorf("expected and actual error lists differ: %s", cmp.Diff(test.expectedErrorList, errList))
+			}
+
+			errStr := ""
+			if agg := errList.ToAggregate(); agg != nil {
+				errStr = agg.Error()
+			}
+			if !reflect.DeepEqual(errStr, test.expectedErrorString) {
+				t.Errorf("expected and actual error strings differ: %s", cmp.Diff(test.expectedErrorString, errStr))
 			}
 		})
 	}
 }
 
 func TestValidateScyllaClusterUpdate(t *testing.T) {
 	tests := []struct {
-		name    string
-		old     *v1.ScyllaCluster
-		new     *v1.ScyllaCluster
-		allowed bool
+		name                string
+		old                 *v1.ScyllaCluster
+		new                 *v1.ScyllaCluster
+		expectedErrorList   field.ErrorList
+		expectedErrorString string
 	}{
 		{
-			name:    "same as old",
-			old:     unit.NewSingleRackCluster(3),
-			new:     unit.NewSingleRackCluster(3),
-			allowed: true,
+			name:              "same as old",
+			old:               unit.NewSingleRackCluster(3),
+			new:               unit.NewSingleRackCluster(3),
+			expectedErrorList: field.ErrorList{},
```

Comment:
nit: also explicitly set `expectedErrorString` so the intent is clear (applies globally)

Reply:
Thanks, missed it. Fixed